sale_order_line_tree_view/views/sale_order_line_views.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.ui.view" id="view_sale_order_line_tree">
+        <field name="name">sale.order.line.tree</field>
+        <field name="model">sale.order.line</field>
+        <field name="arch" type="xml">
+            <tree string="Sales Orders Lines">
+                <field name="order_id"/>
+                <field name="order_partner_id"/>
+                <field name="categ_id"/>
+                <field name="name"/>
+                <field name="lot_id"/>
+                <field name="salesman_id"/>
+                <field name="product_uom_qty"/>
+                <field name="qty_delivered"/>
+                <field name="qty_invoiced"/>
+                <field name="qty_to_invoice"/>
+                <field name="price_subtotal" sum="Total" widget="monetary"/>
+                <field name="state"/>
+            </tree>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="view_sale_order_line_search">
+        <field name="name">sale.order.line.search</field>
+        <field name="model">sale.order.line</field>
+        <field name="arch" type="xml">
+            <search string="Search options">
+                <filter string="To Be Delivered"
+                        domain="[('to_be_delivered','=',True)]"/>
+                <filter string="Sales"
+                        domain="[('state','in',('sale','done'))]"/>
+                <separator/>
+                <field name="categ_id"/>
+                <field name="order_partner_id"/>
+                <group expand="0" string="Group By">
+                    <filter string="Expected Date"
+                            domain="[]"
+                            context="{'group_by':'expected_date'}"/>
+                </group>
+            </search>
+        </field>
+    </record>
+
+    <record id="action_sale_order_line" model="ir.actions.act_window">
+        <field name="name">Sales Orders Lines</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">sale.order.line</field>
+        <field name="view_id" ref="view_sale_order_line_tree"/>
+        <field name="search_view_id" ref="view_sale_order_line_search"/>
+        <field name="domain">[('is_delivery','!=',True)]</field>
```

Comment:
let's show these records as well rather than hiding them. We aim to make the function to be rather generic.